core/Pos/Crypto/Signing.hs

```diff
@@ -170,6 +171,12 @@ fullSignatureHexF :: Format r (Signature a -> r)
 fullSignatureHexF = later $ \(Signature x) ->
     B16.formatBase16 . CC.unXSignature $ x
 
+-- | Parse 'Signature' from base16 encoded string.
+parseFullSignature :: Text -> Maybe (Signature a)
```

Comment:
Why does it return `Maybe` instead of `Either` with error message? It doesn't make sense to me.

Reply:
lol, I just copied parser for public key

Reply:
will fix them

Reply:
Parser for public key was strange too.